pkg/processor/worker.go

```diff
@@ -301,63 +302,67 @@ func (wrk *worker) checkNeedsRebuild() bool {
 
 // updateResource checks if actual resource satisfies the desired spec.
 // Returns non-nil object with updates applied or nil if actual matches desired.
-func (wrk *worker) updateResource(desired *smith.Resource, actual *unstructured.Unstructured) (*unstructured.Unstructured, error) {
+func updateResource(scheme *runtime.Scheme, desired *smith.Resource, actual *unstructured.Unstructured) (*unstructured.Unstructured, error) {
 	spec := desired.Spec
 
 	// TODO Handle deleted or to be deleted object. We need to wait for the object to be gone.
 
-	upd, err := wrk.tp.scheme.Copy(actual)
+	upd, err := scheme.Copy(actual)
 	if err != nil {
 		return nil, err
 	}
 	updated := upd.(*unstructured.Unstructured)
+	delete(updated.Object, "status")
+
+	actClone, err := scheme.Copy(updated)
+	if err != nil {
+		return nil, err
+	}
+	actualClone := actClone.(*unstructured.Unstructured)
+
+	// This is to ensure those fields actually exist in underlying map whether they are nil or empty slices/map
+	actualClone.SetKind(actualClone.GetKind())
```

Comment:
At this point in code `actual` == `updated` == `actualClone`. I used the same object to make it more explicit that copying fields in not the purpose of the code.